src/tlo/util.py

```diff
@@ -436,3 +437,30 @@ def get_person_id_to_inherit_from(child_id, mother_id, population_dataframe, rng
         return abs(mother_id)
     elif mother_id >= 0:
         return mother_id
+
+
+def read_csv_files(resource_filepath: Path, file_name: list[str] = None) -> DataFrame | dict[str, DataFrame]:
+    """
+    A function to read csv files in a similar way pandas read Excel files(pd.read_excel).
+
+    :param resource_filepath:  path to resource file folder
+    :param file_name: preferred csv file name(s). This is the same as sheet names in Excel file
+
+    """
+    all_data = {}
+    if len(file_name) == 1:
+        df = pd.read_csv(f'{resource_filepath}/{file_name[0]}.csv')
+        return df.drop(df.filter(like='Unnamed'), axis=1)
+
+    elif file_name is not None:
+        for f_name in file_name:
+            df = pd.read_csv(f'{resource_filepath}/{f_name}.csv')
+            all_data[f_name] = df.drop(df.filter(like='Unnamed'), axis=1)
+
+    else:
+        for f_name in resource_filepath.rglob("*.csv"):
+            file_base_name = str(f_name).split('/')[-1].split('.')[0]  # Extract base name without extension
+            df = pd.read_csv(f_name)
+            all_data[file_base_name] = df.drop(df.filter(like='Unnamed'), axis=1)
```

Comment:
In addition to @tbhallett's comment regarding documenting this behaviour, ideally we would avoid the reperition of the `df.drop(df.filter(like='Unnamed'), axis=1)` snippet - either by refactoring the conditional structure here or factoring out into a helper function, so that its easier to update and keep consistent.